src/pages/addCard/hooks/useCardNumberInput.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback } from "react";
+
+export default function useCardNumberInput() {
+  const toMaskedCardNumber = useCallback(
+    (value: string) =>
+      value
+        .split("")
+        .map((str, index) => {
+          const displayedStr = index > 7 ? "*" : str;
+          return index !== 0 && index % 4 === 0
+            ? `-${displayedStr}`
+            : displayedStr;
+        })
+        .join(""),
+    []
+  );
+
+  return { toMaskedCardNumber };
+}
```

Comment:
```suggestion
export default function useCardNumberInput() {
  return = useCallback(
    (value: string) =>
      value
        .split("")
        .map((str, index) => {
          const displayedStr = index > 7 ? "*" : str;
          return index !== 0 && index % 4 === 0
            ? `-${displayedStr}`
            : displayedStr;
        })
        .join(""),
    []
  );
}
```

차라리 이렇게 하면 사용하는 측에서 함수 이름을 더 유연하게 정의할 수 있어서 좋지 않을까요~?

Reply:
이렇게 하는 방식은 생각해보지 않았는데, 좋은 의견 감사합니다!!!